lib/router.js

```diff
@@ -108,11 +108,15 @@ function match (router, method, route, lastMatch) {
     if (value && value[IS_ROUTER_SYM]) {
       // if there's nothing left of the path, pretend
       // it's a slash so we can pick up nested / routes
-      const remainingPath = rest || '/'
+      const remainingPath = (
+          target.isConcat
+          ? `/${rest}`
+          : rest || '/'
+        )
```

Comment:
🔴 The three cases are:

1. If we're actually _concatenating_ routes, don't consume the slash from the incoming route. Preserve it.
2. If we're nesting and we haven't consumed the rest of the url, send it through.
3. If we're nesting and we _have_ consumed the rest of the url, simulate a final slash. This lets the nested router mount a route at, e.g., `GET / bar`, such that a toplevel `* /nested foo` pointing to it would route `GET /nested` to `bar`.